src/main/java/io/anserini/embeddings/search/IndexW2V.java

```diff
@@ -0,0 +1,177 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.anserini.embeddings.search;
+
+import io.anserini.index.generator.LuceneDocumentGenerator;
+import io.anserini.rerank.ScoredDocuments;
+import io.anserini.util.AnalyzerUtils;
+import org.apache.commons.io.output.ByteArrayOutputStream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.analysis.core.WhitespaceAnalyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.StoredField;
+import org.apache.lucene.document.StringField;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.store.FSDirectory;
+import org.kohsuke.args4j.*;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.FloatBuffer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+
+import static io.anserini.index.generator.LuceneDocumentGenerator.FIELD_BODY;
+import static io.anserini.index.generator.LuceneDocumentGenerator.FIELD_ID;
+
+public final class IndexW2V {
+  private static final Logger LOG = LogManager.getLogger(IndexW2V.class);
+
+  public static final class Args {
+
+    // required arguments
+    @Option(name = "-input", metaVar = "[Path]", required = true, usage = "collection path")
+    public String input;
+
+    @Option(name = "-index", metaVar = "[Path]", required = true, usage = "index path")
+    public String index;
+
+    @Option(name = "-term", metaVar = "String", usage = "get the embeddings corresponding to the term")
+    public String term = "";
+
+    //optional arguments
+    @Option(name = "--build", usage = "builds index if specified")
+    public boolean build = false;
+  }
+
+  private final IndexW2V.Args args;
+  private final Path indexPath;
+  private final FSDirectory directory;
+
+  public IndexW2V(IndexW2V.Args args) throws Exception {
+    this.args = args;
+    this.indexPath = Paths.get(args.index);
+
+    if (!Files.exists(this.indexPath)) {
+      Files.createDirectories(this.indexPath);
+    }
+
+    this.directory = FSDirectory.open(indexPath);
+  }
+
+  public float[] getEmbeddings(String term) throws IOException {
+    DirectoryReader reader = DirectoryReader.open(directory);
```

Comment:
Should I create yet another file, maybe? Just for obtaining the embeddings? 
Also, I could initialize the reader in a constructor, but the index may not exist... Should separate the logic by creating two different files?